hw/opentitan/ot_i2c.c

```diff
@@ -1238,13 +1239,33 @@ static void ot_i2c_target_send_async(I2CSlave *target, uint8_t data)
     }
 }
 
+static bool ot_i2c_target_match_and_add(I2CSlave *candidate, uint8_t address,
+                                        bool broadcast,
+                                        I2CNodeList *current_devs)
+{
+    BusState *abus = qdev_get_parent_bus(DEVICE(candidate));
+    OtI2CState *s = OT_I2C(abus->parent);
+
+    /* Check address, subject to address masking. */
+    if ((address & s->address_mask) == candidate->address || broadcast) {
```

Comment:
https://opentitan.org/book/hw/ip/i2c/doc/registers.html#target_id

> I2C target mask number 0. At least one bit in MASK0 must be set to 1 for ADDRESS0 to be used.

I think that this should be


```suggestion
    if (s->address_mask && (address & s->address_mask) == candidate->address || broadcast) {
```